src/common/window/session-storage.js

```diff
@@ -6,34 +6,68 @@
  * @environment Browser script
  */
 import { generateRandomHexString } from '../ids/unique-id'
-import { isBrowserScope } from '../util/global-scope'
 
-export { getCurrentSessionIdOrMakeNew }
-
-const SESS_ID = 'NRBA_SESSION_ID' // prevents potential key collisions in session storage
+const AGENT_PREFIX = 'NRBA/'
 
 /**
- * @returns {string} This tab|window's session identifier, or a new ID if not found in storage
+ * @param {string} key
+ * @param {string} value
+ * @returns true, if item was set successfully. false, if storage is full or setItem threw an exception.
  */
-function getCurrentSessionIdOrMakeNew () {
-  if (!isBrowserScope) {
-    return null
+export function putInBrowserStorage (key, value, subgroup = '') {
```

Comment:
I am honestly debating if it merits making a storage wrapper or using something like [localforage](https://github.com/localForage/localForage).

Reply:
Was having the same thoughts. Session replay May end up changing our definitions of how we manage a session so we should iterate on this concept. I don't want to bloat the scope here but it's on the to-do list